docs/core/utilities.md

```diff
@@ -0,0 +1,256 @@
+      * [Example](#example-6)
+  * [ContentType](#contenttype)
+    * [encodeContentType](#encodecontenttype)
+      * [Example](#example-7)
+    * [decodeContentType](#decodecontenttype)
+      * [Example](#example-8)
+    * [isContentTypeValid](#iscontenttypevalid)
+      * [Example](#example-9)
+  * [Transaction](#transaction)
+    * [waitForTransaction](#waitfortransaction)
+      * [Example](#example-10)
+<!-- TOC -->
+
+## SporeConfig
+
+### getSporeConfig
+
+Get the global default SporeConfig.
+The default config is "predefinedSporeConfigs.Aggron4".
+
+```typescript
+declare function getSporeConfig<T extends string = string>(): SporeConfig<T>;
+```
+
+#### Example
+
+```typescript
+const config = getSporeConfig();
+```
+
+### setSporeConfig
+
+Set the global default SporeConfig.
+The default config is "predefinedSporeConfigs.Aggron4".
+
+```typescript
+declare function setSporeConfig<T extends string = string>(config: SporeConfig<T>): void; 
+```
+
+#### Example
+
+```typescript
+setSporeConfig(predefinedSporeConfigs.Aggron4);
+```
+
+### forkSporeConfig
+
+Clone and create a new SporeConfig.
+
+```typescript
+declare function forkSporeConfig<T1 extends string, T2 extends string>(
+  origin: SporeConfig<T1>,
+  change: Partial<SporeConfig<T2>>,
+): SporeConfig<T1 | T2>;
+```
+
+#### Example
+
+```typescript
+const originalConfig = predefinedSporeConfigs.Aggron4;
+const forkedConfig = forkSporeConfig(originalConfig, {
+  maxTransactionSize: 100,
+});
+```
+
+## SporeScript
+
+### getSporeScript
+
+Get a specific SporeScript from SporeConfig by "scriptName" with optional "scriptId" or "tags".
+Throws an error if the script doesn't exist.
+
+```typescript
+declare function getSporeScript(config: SporeConfig, scriptName: string): SporeScript;
+declare function getSporeScript(config: SporeConfig, scriptName: string, tags: string[]): SporeScript;
+declare function getSporeScript(config: SporeConfig, scriptName: string, scriptId: ScriptId): SporeScript;
+```
+
+#### Example
+
+Getting the latest version of the "Spore" script:
+
+```typescript
+const sporeScript = getSporeScript(config, 'Spore');
+```
+
+Getting the latest "Spore" script with the "latest" tag:
+
+```typescript
+const sporeScript = getSporeScript(config, 'Spore', ['latest']);
+```
+
+Getting the exact "Spore" script with the specified ScriptId:
+
+```typescript
+const sporeScript = getSporeScript(config, 'Spore', {
+  codehash: '0xbbad126377d45f90a8ee120da988a2d7332c78ba8fd679aab478a19d6c133494',
+  hashType: 'data1',
+});
+```
+
+### isSporeScriptSupported
+
+Returns a boolean indicating weather if the target ScriptId exists in the SporeConfig.
+If "scriptName" is passed, it also checks whether the name of the target script matches.
+
+```typescript
+declare function isSporeScriptSupported(config: SporeConfig, scriptId: ScriptId, scriptName?: string): boolean;
+```
+
+#### Example
+
+Check if the target cell is any script supported by the SporeConfig:
```

Comment:
edit, grammar: "Check if the target cell is a script supported by the SporeConfig:"